integration/delete_compat_test.go

```diff
@@ -41,18 +41,55 @@ func TestDeleteCompat(t *testing.T) {
 			filters:    []bson.D{},
 			resultType: emptyResult,
 		},
+		"OrderedTrueNoError": {
+			filters: []bson.D{
+				{{"_id", "string"}},
+				{{"_id", "double"}},
+			},
+			ordered: true,
+		},
+		"OrderedFalseNoError": {
+			filters: []bson.D{
+				{{"_id", "string"}},
+				{{"_id", "double"}},
+			},
+			ordered: false,
+		},
+		"OrderedTrue": {
+			filters: []bson.D{
+				{{"_id", "string"}},
+				{{"v", bson.D{{"$all", 9}}}},
+				{{"_id", "double"}},
+			},
+			ordered: true,
+		},
+		"OrderedFalse": {
+			filters: []bson.D{
+				{{"_id", "string"}},
+				{{"v", bson.D{{"$all", 9}}}},
+				{{"_id", "double"}},
+			},
+			ordered: false,
+		},
+		"OrderedFalseTwoErrors": {
```

Comment:
I suggest adding two more cases:
 - Something like `OrderedFalseAllErrors` - when all the statements are not correct.
 - Something like `OrderedTrueAllErrors` - same but for `ordered: true`.